docs/source/transaction_family_specifications/identity_transaction_family.rst

```diff
@@ -0,0 +1,217 @@
+      string key = 2;
+
+    }
+
+    // name of the policy, this should be unique.
+    string name = 1;
+
+    // The entries will be processed in order from first to last.
+    repeated Entry entries = 2;
+  }
+
+Roles
+-----
+A role will be made up of a role name and the name of the policy to be enforced
+for that role. The data will be stored in state at the address described above
+using the following protobuf messages:
+
+.. code-block:: protobuf
+
+  // Roles will be stored in a RoleList to handle state collisions. The Roles
+  // will be sorted by name.
+  message RoleList {
+    repeated Role roles = 1;
+  }
+
+  message Role{
+    // Role name
+    string name = 1;
+
+    // Name of corresponding policy
+    string policy_name = 2;
+  }
+
+Addressing
+----------
+All identity data will be stored under the special namespace of “00001d”.
+
+For each policy, the address will be formed by concatenating the namespace, the
+special policy namespace of “00”, and the first 62 characters of the SHA-256
+hash of the policy name:
+
+.. code-block:: pycon
+
+ >>> "00001d" + "00" + hashlib.sha256(policy_name.encode()).hexdigest()[:62]
+
+Address construction for roles will follow a pattern similar to address
+construction in the settings namespace. Role names will be broken into four
+parts, where parts of the string are delimited by the "." character. For
+example, the key a.b.c would be split into the parts "a", "b", "c", and the
+empty string. If a key would have more than four parts the extra parts are left
+in the last part. For example, the key a.b.c.d.e would be split into "a", "b",
+"c", and "d.e".
+
+A short hash is computed for each part. For the first part the first 14
+characters of the SHA-256 hash are used. For the remaining parts the first 16
+characters of the SHA-256 hash are used. The address is formed by concatenating
+the identity namespace “00001d”, the role namespace “01”, and the four short
+hashes.
+
+For example, the address for the role client.query_state would be constructed
+as follows:
+
+.. code-block:: pycon
+
+  >>> "00001d"+ "01" + hashlib.sha256('client'.encode()).hexdigest()[:14]+ \
+    hashlib.sha256('query_state'encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]
+
+Transaction Payload
+===================
+Identity transaction family payloads are defined by the following protocol
+buffers code:
+
+File: sawtooth-core/families/identity/protos/identity.proto
+
+.. code-block:: protobuf
+
+  message IdentityPayload {
+      enum IdentityType {
+        POLICY = 0;
+        ROLE = 1;
+      }
+
+      // Which type of payload this is for
+      IdentityType type = 1;
+
+      // Serialize bytes of a role or a policy
+      bytes data = 2;
+  }
+
+Transaction Header
+==================
+
+Inputs and Outputs
+------------------
+
+The inputs for config family transactions must include:
```

Comment:
Config family? Shouldn't that be Settings Family or Identity Family (not sure which we are referring to here)? Also, suggest capitalizing the name.